src/Features/CSharp/Portable/Snippets/AbstractCSharpForLoopSnippetProvider.cs

```diff
@@ -56,7 +56,7 @@ protected override SyntaxNode GenerateStatement(SyntaxGenerator generator, Synta
             }
             else
             {
-                var inlineExpressionType = semanticModel.GetTypeInfo(inlineExpression).Type;
+                var inlineExpressionType = semanticModel.GetTypeInfo(OriginalInlineExpression!).Type;
```

Comment:
📝 This seems like a burdensome requirement, which is avoided by my suggested change. In general, inline snippets seems like a strong contender for a future public API and passing in the original node seems easier all around.